src/pieces/settings.py

```diff
@@ -53,21 +53,34 @@ def get_model(cls):
 			return cls._model_name
 
 		model_id = cls.get_from_pickle(cls.models_file,"model_id")
-		models_reverse = {v:k for k,v in cls.pieces_client.get_models().items()}
-		cls._model_name = models_reverse.get(model_id)
+		if model_id:
+			models_reverse = {v:k for k,v in cls.pieces_client.get_models().items()}
```

Comment:
i'm a little confused why we have to invert the dictionary here, why would the value be the id in the first place?

Reply:
This is the wrapper architecture everything in the wrapper uses the model name but here I need the id